destination_middleware.go

```diff
@@ -425,26 +425,22 @@ type destinationWithSchemaExtraction struct {
 	Destination
 	config *DestinationWithSchemaExtraction
 
-	payloadEnabled bool
-	keyEnabled     bool
-
 	payloadWarnOnce sync.Once
 	keyWarnOnce     sync.Once
 }
 
 func (d *destinationWithSchemaExtraction) Write(ctx context.Context, records []opencdc.Record) (int, error) {
-	if d.keyEnabled {
-		for i := range records {
+	for i := range records {
```

Comment:
The change here is about using a single loop for all the records, where we decode the key and/or payload in each iteration. Previously, we used two loops: one to decode the keys, and one to decode the payloads.